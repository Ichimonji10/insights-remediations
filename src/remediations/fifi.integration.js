'use strict';

const { request, auth } = require('../test');
const utils = require('../middleware/identity/utils');

describe('FiFi', function () {
    describe('connection status', function () {
        test('obtains connection status', async () => {
            const {text} = await request
            .get('/v1/remediations/0ecb5db7-2f1a-441b-8220-e5ce45066f50/connection_status?pretty')
            .set(auth.fifi)
            .expect(200);

            expect(text).toMatchSnapshot();
        });

        test('404s on empty playbook', async () => {
            const {body} = await request
            .get('/v1/remediations/249f142c-2ae3-4c3f-b2ec-c8c5881f6927/connection_status?pretty')
            .set(auth.fifi)
            .expect(200);

            body.meta.count.should.eql(0);
            body.meta.total.should.eql(0);
            body.data.should.eql([]);
        });

        test('400 get connection status', async () => {
            await request
            .set(auth.fifi)
            .get('/v1/remediations/66eec356-dd06-4c72-a3b6-ef27d150000/connection_status')
            .expect(400);
        });

        test('get connection status with false smartManagement', async () => {
            await request
            .get('/v1/remediations/0ecb5db7-2f1a-441b-8220-e5ce45066f50/connection_status')
            .set(utils.IDENTITY_HEADER, utils.createIdentityHeader('fifi', 'fifi', true, data => {
                data.entitlements.smart_management = false;
                return data;
            }))
            .expect(403);
        });

        test('sets ETag', async () => {
            const {headers} = await request
            .get('/v1/remediations/0ecb5db7-2f1a-441b-8220-e5ce45066f50/connection_status?pretty')
            .set(auth.fifi)
            .expect(200);

            headers.etag.should.equal('"1062-Pl88DazTBuJo//SQVNUn6pZAllk"');
        });

        test('304s on ETag match', async () => {
            await request
            .get('/v1/remediations/0ecb5db7-2f1a-441b-8220-e5ce45066f50/connection_status?pretty')
            .set(auth.fifi)
            .set('if-none-match', '"1062-Pl88DazTBuJo//SQVNUn6pZAllk"')
            .expect(304);
        });
    });

    describe('playbook run', function () {
        test('execute playbook run', async () => {
            await request
            .post('/v1/remediations/0ecb5db7-2f1a-441b-8220-e5ce45066f50/playbook_runs')
            .set(auth.fifi)
            .expect(204);
        });

        test('execute playbook run with false smartManagement', async () => {
            await request
            .post('/v1/remediations/0ecb5db7-2f1a-441b-8220-e5ce45066f50/playbook_runs')
            .set(utils.IDENTITY_HEADER, utils.createIdentityHeader('fifi', 'fifi', true, data => {
                data.entitlements.smart_management = false;
                return data;
            }))
            .expect(403);
        });
    });
});
