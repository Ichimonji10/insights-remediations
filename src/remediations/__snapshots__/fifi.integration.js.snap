// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiFi connection status obtains connection status 1`] = `
"{
    \\"meta\\": {
        \\"count\\": 6,
        \\"total\\": 6
    },
    \\"data\\": [
        {
            \\"executor_id\\": \\"01bf542e-6092-485c-ba04-c656d77f988a\\",
            \\"executor_type\\": \\"satellite\\",
            \\"executor_name\\": \\"Satellite 01bf542e-6092-485c-ba04-c656d77f988a\\",
            \\"system_count\\": 4,
            \\"connection_status\\": \\"no_source\\"
        },
        {
            \\"executor_id\\": \\"722ec903-f4b5-4b1f-9c2f-23fc7b0ba390\\",
            \\"executor_type\\": \\"satellite\\",
            \\"executor_name\\": \\"Satellite 1 (connected)\\",
            \\"system_count\\": 3,
            \\"connection_status\\": \\"connected\\"
        },
        {
            \\"executor_id\\": \\"409dd231-6297-43a6-a726-5ce56923d624\\",
            \\"executor_type\\": \\"satellite\\",
            \\"executor_name\\": \\"Satellite 2 (disconnected)\\",
            \\"system_count\\": 6,
            \\"connection_status\\": \\"disconnected\\"
        },
        {
            \\"executor_id\\": \\"72f44b25-64a7-4ee7-a94e-3beed9393972\\",
            \\"executor_type\\": \\"satellite\\",
            \\"executor_name\\": \\"Satellite 3 (no receptor configured)\\",
            \\"system_count\\": 5,
            \\"connection_status\\": \\"no_receptor\\"
        },
        {
            \\"executor_id\\": \\"63142926-46a5-498b-9614-01f2f66fd40b\\",
            \\"executor_type\\": \\"satellite\\",
            \\"executor_name\\": \\"Satellite 4 (connected)\\",
            \\"system_count\\": 1,
            \\"connection_status\\": \\"connected\\"
        },
        {
            \\"executor_id\\": null,
            \\"executor_type\\": null,
            \\"executor_name\\": null,
            \\"system_count\\": 2,
            \\"connection_status\\": \\"no_executor\\"
        }
    ]
}"
`;

exports[`FiFi playbook run check object being send to receptor connector 1`] = `
Array [
  Object {
    "account": "fifi",
    "directive": "receptor_satellite:execute",
    "payload": Object {
      "account": "fifi",
      "config": Object {
        "fifi_text_update_full": true,
        "fifi_text_update_interval": 5000,
        "fifi_text_updates": true,
      },
      "hosts": Array [
        "355986a3-5f37-40f7-8f36-c3ac928ce190.example.com",
        "b84f4322-a0b8-4fb9-a8dc-8abb9ee16bc0",
        "d5174274-4307-4fac-84fd-da2c3497657c",
      ],
      "playbook": "---
# Red Hat Insights has recommended one or more actions for you, a system administrator, to review and if you
# deem appropriate, deploy on your systems running Red Hat software. Based on the analysis, we have automatically
# generated an Ansible Playbook for you. Please review and test the recommended actions and the Playbook as
# they may contain configuration changes, updates, reboots and/or other changes to your systems. Red Hat is not
# responsible for any adverse outcomes related to these recommendations or Playbooks.
#
# FiFI playbook 3
# https://cloud.redhat.com/insights/remediations/249f142c-2ae3-4c3f-b2ec-c8c5881f8561
# Generated by Red Hat Insights on Sat, 29 Dec 2018 08:20:35 GMT

# Fixes test:ping
# Identifier: (test:ping,fix)
# Version: unknown
- name: ping
  hosts: \\"355986a3-5f37-40f7-8f36-c3ac928ce190.example.com,b84f4322-a0b8-4fb9-a8dc-8abb9ee16bc0,d5174274-4307-4fac-84fd-da2c3497657c\\"
  tasks:
    - ping

- name: run insights
  hosts: \\"355986a3-5f37-40f7-8f36-c3ac928ce190.example.com,b84f4322-a0b8-4fb9-a8dc-8abb9ee16bc0,d5174274-4307-4fac-84fd-da2c3497657c\\"
  become: True
  gather_facts: False
  tasks:
    - name: run insights
      command: insights-client
      changed_when: false",
      "playbook_run_id": "249f142c-2ae3-4c3f-b2ec-c8c588999999",
      "remediation_id": "249f142c-2ae3-4c3f-b2ec-c8c5881f8561",
      "remediation_name": "FiFI playbook 3",
    },
    "recipient": "098765",
  },
]
`;

exports[`FiFi playbook run check object being send to receptor connector 2`] = `
Array [
  Object {
    "account": "fifi",
    "directive": "receptor_satellite:execute",
    "payload": Object {
      "account": "fifi",
      "config": Object {
        "fifi_text_update_full": true,
        "fifi_text_update_interval": 5000,
        "fifi_text_updates": true,
      },
      "hosts": Array [
        "35e9b452-e405-499c-9c6e-120010b7b465.example.com",
      ],
      "playbook": "---
# Red Hat Insights has recommended one or more actions for you, a system administrator, to review and if you
# deem appropriate, deploy on your systems running Red Hat software. Based on the analysis, we have automatically
# generated an Ansible Playbook for you. Please review and test the recommended actions and the Playbook as
# they may contain configuration changes, updates, reboots and/or other changes to your systems. Red Hat is not
# responsible for any adverse outcomes related to these recommendations or Playbooks.
#
# FiFI playbook 3
# https://cloud.redhat.com/insights/remediations/249f142c-2ae3-4c3f-b2ec-c8c5881f8561
# Generated by Red Hat Insights on Sat, 29 Dec 2018 08:20:35 GMT

# Fixes test:ping
# Identifier: (test:ping,fix)
# Version: unknown
- name: ping
  hosts: \\"35e9b452-e405-499c-9c6e-120010b7b465.example.com\\"
  tasks:
    - ping

- name: run insights
  hosts: \\"35e9b452-e405-499c-9c6e-120010b7b465.example.com\\"
  become: True
  gather_facts: False
  tasks:
    - name: run insights
      command: insights-client
      changed_when: false",
      "playbook_run_id": "249f142c-2ae3-4c3f-b2ec-c8c588999999",
      "remediation_id": "249f142c-2ae3-4c3f-b2ec-c8c5881f8561",
      "remediation_name": "FiFI playbook 3",
    },
    "recipient": "098768",
  },
]
`;
